### PR TITLE
feat: implement createToken function in authservice

### LINKED ISSUE #34: [Auth] create a vaild token
**Subject**
JWT

**Detail**
I added createToken() which creates a valid token when it receives login request.

**Reference**

### PR DESCRIPTION
**Pull request by sojeongw**
## Related Issue

resolve: #34

## Description

The createToken() creates a valid token when it receives a login request.

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] Test case
- [x] End of Work

### CODE REVIEW

File: server/src/auth/auth.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AuthService } from './auth.service';
+import { JwtService } from '@nestjs/jwt';
+import { anything, instance, mock, when } from 'ts-mockito';
+
+describe('AuthService', () => {
+  let service: AuthService;
+  const mockJwtService: JwtService = mock(JwtService);
+
+  describe('#di()', () => {
```
Comment (ttkmw) [inline server/src/auth/auth.service.spec.ts:10]: I think #function() is convention for function. There is no di function so i think removing convention is better

File: server/src/auth/auth.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      const accessToken = 'accesstoken';
+
+      const result = {
+        expiresIn: 3600,
+        accessToken,
+      };
+
+      when(mockJwtService.sign(anything())).thenReturn(accessToken);
+      service = new AuthService(instance(mockJwtService));
+      expect(service.createToken(email)).toEqual(result);
```
Comment (ttkmw) [inline server/src/auth/auth.service.spec.ts:36]: Isnt theres fail case?

File: server/src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+    constructor(
+        @Inject('JwtService') private readonly jwtService: JwtService,
+    ) {}
+
+    createToken(email: string) {
+        const user: JwtPayload = { email };
```
Comment (ttkmw) [inline server/src/auth/auth.service.ts:12]: I think user: payload is awkward.
payload: Payload or
user:User isnt better?

Comment (ttkmw) [reply server/src/auth/auth.service.ts:12]: There is no email field in our user object.

I think address is needed

File: server/src/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { JwtPayload } from './interfaces/jwt-payload.interface';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+    constructor(
+        @Inject('JwtService') private readonly jwtService: JwtService,
```
Comment (ttkmw) [inline server/src/auth/auth.service.ts:8]: I think registering jwtservice is needed

File: server/src/auth/auth.service.ts (hunk 3 above)
Comment (hihiboss) [reply server/src/auth/auth.service.ts:12]: @inDlife 
I think `email` is a variable which is parameter.

Comment (hihiboss) [reply server/src/auth/auth.service.ts:12]: I recommend an idea `const userPayload: UserPayload` and `interface UserPayload`.

File: server/src/auth/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    ) {}
+
+    createToken(email: string) {
```
Comment (hihiboss) [inline server/src/auth/auth.service.ts:11]: Write return type!

File: server/src/auth/auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        const user: JwtPayload = { email };
+        const accessToken = this.jwtService.sign(user);
+        const seconds = 3600;
+        return {
+            expiresIn: seconds,
+            accessToken,
```
Comment (hihiboss) [inline server/src/auth/auth.service.ts:17]: In here, it seems like to need a field name for `accessToken`.

File: server/src/auth/interfaces/jwt-payload.interface.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export interface JwtPayload {
```
Comment (hihiboss) [inline server/src/auth/interfaces/jwt-payload.interface.ts:1]: Is this interface? Not entity like struct?

**Review by hihiboss — COMMENTED** [review]
Thank you for your first contribution! :)

File: server/src/auth/auth.service.spec.ts (hunk 2 above)
Comment (sojeongw) [reply server/src/auth/auth.service.spec.ts:36]: @inDlife JwtService.sign() is part of JWT library which guarantees returning a vaild token after validate(). I think sign() doesn't need any fail case if validate() works correctly. And as you know, validate() is your task! :-P

**Pull request closed without merging**